src/webview/java/org/medicmobile/webapp/mobile/EmbeddedBrowserActivity.java

```diff
@@ -78,6 +80,8 @@ public void onReceiveValue(String result) {
 	private PhotoGrabber photoGrabber;
 	private SmsSender smsSender;
 
+	private boolean isMigrationRunning = false;
```

Comment:
With this as a boolean it'll reset to false on restart. This may be an issue if it crashes or the user closes it. Can we store it somewhere persistent instead?

Reply:
Yes, but solve that means make the migration process more resilient, with the ability to resume in the cases you mention, but it is out of scope I think for this ticket that only intent to improve the UI so users are not presented to the login page which cause many problems described in the ticket.

Currently, If the app crashes but the migration was finished, there won't be a problem, because the app will be restarted and with our without this patch the UI will load the CHT dashboard because a that point the cookies will be available again. If the crash happens during the migration, is likely the data will be corrupted and the user will need to reinstall the app losing the data, but again this PR cannot address that now.

Reply:
I'm worried about edge cases like... What if the user spawns a [second instance](https://github.com/medic/medic-android/issues/122) of this Activity? What happens if two migrations run in parallel, or one after the other?

However adding some sort of persistent flag is also risky... what if it gets set to `migrating = true` and then the migration fails? It may never recover.

This is going to have to be tested really thoroughly to avoid bricking or corrupting phones. Are you confident this is the safest approach?